File: src/main/java/workshop/java/_02_java_interop/Position.java
package workshop.java._02_java_interop;

public class Position {

    public static final Position TOP_LEFT_CORNER = new Position(0, 0);

    public static final Position UNKNOWN = new Position(Integer.MAX_VALUE, Integer.MAX_VALUE);

    private final int column;
    private final int row;

    private Position(int column, int row) {
        this.column = column;
        this.row = row;
    }

    public int getColumn() {
        return column;
    }

    public int getRow() {
        return row;
    }

    /**
     * Factory method for [Position].
     */
    public static Position of(int column, int row) {
        return new Position(column, row);
    }

}

File: src/main/kotlin/workshop/kotlin/_02_java_interop/Position.kt
package workshop.kotlin._02_java_interop

/**
 * You can check this class in its original version at:
 * https://github.com/Hexworks/zircon/blob/master/src/main/kotlin/org/codetome/zircon/api/Position.kt
 */
data class Position(val column: Int,
                    val row: Int) {


    companion object {

        // TODO: fill in the missing elements from the java `Position`
        // TODO: use @JvmField and @JvmStatic

        /**
         * Constant for the top-left corner (0x0)
         */
        @JvmField
        val TOP_LEFT_CORNER = Position(0, 0)

    }
}